### PR TITLE
GCI-backed masters mount srv/kubernetes and srv/sshproxy in the right place

### LINKED ISSUE #26235: Master PD file location inconsistent between Debian & GCI
The location of apiserver auth certs and keys and the kube-apiserver SSH keys are inconsistent between Debian and GCI.  This means that master upgrades from Debian to GCI are broken.

In Debian, Salt puts these files in `master-pd/srv/kubernetes` (via `/srv/salt-overlay/salt/kube-apiserver/basic_auth.csv` and `/srv/kubernetes/basic_auth.csv`, see [configure-vm.sh](https://github.com/kubernetes/kubernetes/blob/1ef19062096c4fcf8ddeaf3a46731d9f262a0c04/cluster/gce/configure-vm.sh) and [init.sls](https://github.com/kubernetes/kubernetes/blob/1ef19062096c4fcf8ddeaf3a46731d9f262a0c04/cluster/saltbase/salt/kube-apiserver/init.sls)), whereas GCI expects it in `master-pd/etc/srv/kubernetes` (see [configure-helper.sh](https://github.com/kubernetes/kubernetes/blob/1ef19062096c4fcf8ddeaf3a46731d9f262a0c04/cluster/gce/gci/configure-helper.sh), originally introduced in https://github.com/kubernetes/kubernetes/pull/20463).

cc @mikedanese @adityakali @andyzheng0831

### PR DESCRIPTION
**Pull request by ikehz**
Fixes #26235.

cc @andyzheng0831

### CODE REVIEW

**Comment by ihmccreery** [comment]
I haven't testing that this fixes the problem completely; I will do so before this should be merged.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -75,16 +75,22 @@ function mount-master-pd {
   mkdir -p "${mount_point}"
   /usr/share/google/safe_format_and_mount -m "mkfs.ext4 -F" "${pd_path}" "${mount_point}" &>/var/log/master-pd-mount.log || \
     { echo "!!! master-pd mount failed, review /var/log/master-pd-mount.log !!!"; return 1; }
+
+  # NOTE: These locations are strongly coupled with locations in
+  # cluster/gce/configure-vm.sh:mount-master-pd, which specifies where files
+  # will exist on the PD for non-GCI masters.  To maintain upgradeability,
+  # these locations must not change.
```
Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: I wouldn't put it as "must not change". AFAIK there is no expectation that switching between os-distros for the same master/node instance (having some state) should always work.

**Comment by adityakali** [comment]
Barring comment, the change looks good to me.

**Comment by ihmccreery** [comment]
@k8s-bot test this again issue #26311

**Comment by ihmccreery** [comment]
@adityakali PTAL.

**Comment by ihmccreery** [comment]
I've tested this manually, and upgrading from 1.2 to this patch seems to work.

**Comment by adityakali** [comment]
LGTM

**Comment by ihmccreery** [comment]
@mikedanese PTAL?  @adityakali isn't a maintainer, so can't technically LGTM.

**Comment by mikedanese** [comment]
LGTM

**Comment by ihmccreery** [comment]
@k8s-bot unit test this again please issue #25539

**Comment by alex-mohr** [comment]
FWIW, leaving this at p1 as it likely unbreaks upgrade tests.

**Comment by alex-mohr** [comment]
as buildcop: marking e2e-not-required due to previously passed tests, apparent low risk, and size.

**Comment by erictune** [comment]
@mikedanese @roberthbailey Does this PR require action by the user when upgrading from 1.2.x to 1.3.0?  (Think about non-developer users.) If so, please edit your first comment to have  a release-note block, like in https://github.com/kubernetes/kubernetes/pull/28132.  If it is just an optional feature, please change the label to just release-note.  If it is not a complete feature by itself, then apply "release-note-none" label instead.

**Comment by roberthbailey** [comment]
No action is required. This PR was specifically made so that the data location wouldn't change during upgrade so that no action would be required.

**Pull request merged**